### ISSUE TITLE
Bug: Unhide field on DependsOn doesn't send value

### ISSUE THREAD

**Issue by muhammadsaeedparacha** [outl1ne/nova-multiselect-field#233]
```
Text::make('Name'),
Multiselect::make('Status')
->options([1 => 'A', 2 => 'B'])
->hide()
->dependsOn(['name'], function ($field, $request, $formData) {
    $field->show();
})
```

This shows the field properly after depends on is triggered, but does not submit it's value on form submission

**Comment by Tarpsvo** [comment]
Heya! Thanks for the PR. :) The fix is now live in version 4.4.2.

**Comment by phlisg** [comment]
Hello!

I've run into a similar bug, this time using the `asyncResource` method instead of `options`.

I updated to the latest `4.4.2`.

With a `dependsOn`, nothing gets sent in the request. Disabling the `dependsOn` makes it work.
Unfortunately I can't switch back to `options` as there are too many models to load.

Tried quickly looking at the PR above and but not sure how `fillIfVisible` is called in this context.

Thank for the help
